leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/PresenceStatus.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2017 RISE SICS AB.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     RISE SICS AB - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.server.queue;
+
+import java.util.concurrent.ScheduledFuture;
+
+/**
+ * Class that contains all the necessary elements to handle the queue mode. Every registration object that uses Queue
+ * Mode has a PresenceStatus object linked to it for handling this mode.
+ */
+
+public class PresenceStatus {
+
+    /* The state of the client: Awake or Sleeping */
+    private Presence state;
+
+    ScheduledFuture<?> clientScheduledFuture;
+
+    private int clientAwakeTime;
+
+    public PresenceStatus() {
+        this.state = Presence.SLEEPING;
+        this.clientAwakeTime = 93000; /* ms, default CoAP value */
```

Comment:
Not easy to change this value with the current implementation. Could be in a parameter of PresenceServiceImpl, and the constructor with argument could be used.